controller/cache/cache.go

```diff
@@ -492,7 +500,7 @@ func (c *liveStateCache) getCluster(server string) (clustercache.ClusterCache, e
 			ref = oldRes.Ref
 		}
 
-		if oldRes != nil && newRes != nil && skipResourceUpdate(resInfo(oldRes), resInfo(newRes)) {
+		if cacheSettings.ignoreResourceUpdatesEnabled && oldRes != nil && newRes != nil && skipResourceUpdate(resInfo(oldRes), resInfo(newRes)) {
```

Comment:
here it uses a old version of cacheSettings defined at line 407 it will get outdated in the callback if the settings change. I think that is why the following code needs to be added. To be validated if this would cause a deadlock

```
			c.lock.RLock()
			cacheSettings := c.cacheSettings
			c.lock.RUnlock()
```